core/math/basis.cpp

```diff
@@ -89,13 +89,26 @@ Basis Basis::orthogonalized() const {
 	return c;
 }
 
+// Returns true of the basis vectors are orthogonal (perpendicular), so it has no skew or shear.
+// See https://en.wikipedia.org/wiki/Orthogonal_basis
 bool Basis::is_orthogonal() const {
-	Basis identity;
-	Basis m = (*this) * transposed();
+	const Vector3 x = get_column(0);
+	const Vector3 y = get_column(1);
+	const Vector3 z = get_column(2);
+	return Math::is_zero_approx(x.dot(y)) && Math::is_zero_approx(x.dot(z)) && Math::is_zero_approx(y.dot(z));
+}
```

Comment:
This is a good point as this is a gotcha not present with the `orthonormal` test.
What would be a good efficient test for one or more of the vectors being zero? Would checking the determinant for zero work? Maybe such a test could be suggested in a comment / doc.

Reply:
> This is a good point as this is a gotcha not present with the `orthonormal` test.

Orthonormal has no such problem, it already ensures all vectors are normalized / unit length (so they can't be zero). It's about orthogonal.

There's also a single similar case for `is_conformal()` as it returns true for a Basis full of zeros (single case up to approximated checks).

> What would be a good efficient test for one or more of the vectors being zero? Would checking the determinant for zero work?

In general no, the determinant can still be zero when no vector is zero (take matrix consisting of the same 9 non-zero values). Implication the other way is true (some zero vector =>(then) determinant is zero).

But if already checked that vectors are orthogonal to each other then yes, zero determinant on top of that would mean there is a zero vector among them. So `is_orthogonal() && !is_zero_approx(determinant())` should do the thing.

Reply:
I don't have a strong opinion of what should happen in the edge case of zero vectors. In #79364 I wrote a similar remark about zero scale:

> Note: Zero scale is invalid but is not validated in master or in this PR. I don't know if that's worth checking for since it's kinda obvious that it's invalid because most things in 3D are considered invalid if they have an zero scale.

I'm open to suggestions on what the desired code should be. Or, alternatively, we could leave it as-is and change it later if a use case comes up (it's not exposed at the moment).

Reply:
The 0 vector is orthogonal to all vectors. To be consistent with this fact, even a matrix with all 0s should return true for this method.